### ISSUE TITLE
write-only variable in binder.ts

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#16418]
**TypeScript Version:**  nightly (2.4.0-dev.20170609)

**Code**

```ts
let root = container;
let hasRest = false;
while (root.parent) {
    if (root.kind === SyntaxKind.ObjectLiteralExpression &&
        root.parent.kind === SyntaxKind.BinaryExpression &&
        (root.parent as BinaryExpression).operatorToken.kind === SyntaxKind.EqualsToken &&
        (root.parent as BinaryExpression).left === root) {
        hasRest = true;
        break;
    }
    root = root.parent;
}
return;
```

**Expected behavior:**

This should do something?

**Actual behavior:**

Writes to two local variables, then returns without using them.

(We should catch this. #11499)

**Comment by andy-ms** [comment]
Just realized the title could come across as harsher then intended...
